tests/functional/backend/SaveQuestionAttributesTest.php

```diff
@@ -0,0 +1,199 @@
+            $attribute2->sendKeys('<script>console.log(1);</script>');
+
+            $savebutton = $web->findElement(WebDriverBy::id('save-and-close-button-create-question'));
+            $savebutton->click();
+
+            $alert = $this->waitForElementShim($web, '#notif-container .alert');
+            $web->wait(10)->until(WebDriverExpectedCondition::visibilityOf($alert));
+
+            self::$testHelper->takeScreenshot(self::$webDriver, __CLASS__ . '_' . __FUNCTION__ . '_afterSave');
+
+        } catch (\Exception $ex) {
+            self::$testHelper->takeScreenshot(self::$webDriver, __CLASS__ . '_' . __FUNCTION__);
+            $this->assertFalse(
+                true,
+                self::$testHelper->javaTrace($ex)
+            );
+        }
+
+        $qid = $survey->groups[0]->questions[0]->qid;
+
+        // Test a filtered attribute
+        $filteredAttribute = \QuestionAttribute::model()->findByAttributes(['qid' => $qid, 'attribute' => 'cssclass']);
+        $this->assertEquals("test-class", $filteredAttribute->value);
+
+        // Test a non-filtered attribute
+        $nonFilteredAttribute = \QuestionAttribute::model()->findByAttributes(['qid' => $qid, 'attribute' => 'nonFilteredAttribute']);
+        $this->assertEquals("<script>console.log(1);</script>", $nonFilteredAttribute->value);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        $oUser = \User::model()->findByPk(self::$newUserId);
+        $oUser->delete();
+    }
+
+    protected function importDemoPlugin()
+    {
+        // Delete demo plugin if it exists
+        $uploadedPluginsDir = \Yii::getPathOfAlias('uploaddir.plugins');
+        $pluginDir = $uploadedPluginsDir . '/NewQuestionAttributesPlugin';
+        if (file_exists($pluginDir)) {
+            \Yii::import('application.helpers.common_helper');
+            rmdirr($pluginDir);
+        }
+
+        $urlMan = \Yii::app()->urlManager;
+        $web = self::$webDriver;
+
+        // Go to plugin manager page
+        $url = $urlMan->createUrl('admin/pluginmanager/sa/index');
+        $web->get($url);
+
+        $button = $this->waitForElementShim($web, '[data-target="#installPluginZipModal"]');
+        $web->wait(10)->until(WebDriverExpectedCondition::elementToBeClickable(WebDriverBy::cssSelector('[data-target="#installPluginZipModal"]')));
+        $button->click();
+
+        // Upload the file
+        $web->wait(10)->until(WebDriverExpectedCondition::elementToBeClickable(WebDriverBy::cssSelector('#installPluginZipModal #the_file')));
+        $fileInput = $web->findByCss('#installPluginZipModal #the_file');
+        $fileInput->setFileDetector(new LocalFileDetector());
+        $file = BASEPATH . '../tests/data/file_upload/NewQuestionAttributesPlugin.zip';
+        $this->assertTrue(file_exists($file));
+        $fileInput->sendKeys($file)->submit();
+
+        $button = $this->waitForElementShim($web, '[type="submit"][value="Install"]');
```

Comment:
@olleharstedt We have problem when executing the test on this line. The button never appears. It seems something is wrong with the previous block of lines, the installation process.

We can't reproduce on our env.
Do you have a magical way of seeing what's happening on the CI?

If not, alternatives:
a- install it through PHP code, not the UI.
b- add it as a demo plugin which can be installable.

ideas?

Reply:
>  Do you have a magical way of seeing what's happening on the CI?

No, that's the sad part of CI. It wasn't possible for Travis, but maybe you can google around for Github Actions?

Reply:
Is it a file/folder permission problem, perhaps?

Reply:
> Is it a file/folder permission problem, perhaps?

I think so. We will do a dump and pick up from there.